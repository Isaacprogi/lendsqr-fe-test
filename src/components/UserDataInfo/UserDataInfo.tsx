import { userDataInfoProps } from '../../utils/types/interface_and_types'


export const UserDataInfo = ({dataValue,dataTitle,dataIdValue, dataIdTitle}:userDataInfoProps) => {
    return (
        <div className='details'>
            <span data-testid={dataIdTitle} className='data-value'>{dataTitle}</span>
            <span data-testid={dataIdValue} className='data-title'>{dataValue}</span>
        </div>
    )
}
