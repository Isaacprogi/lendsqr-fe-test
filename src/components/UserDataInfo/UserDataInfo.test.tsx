import { render, screen } from '@testing-library/react';
import {UserDataInfo} from './UserDataInfo'
import renderer from 'react-test-renderer'


describe('test for NavBar component', ()=>{   
    test('props shoud be present', ()=>{
        render(<UserDataInfo dataIdTitle='dataIdTitle' dataTitle='dataTitle' dataValue='dataValue'  dataIdValue='dataIdValue'   />)
        expect(screen.getByTestId('dataIdTitle').textContent).toBe('dataTitle')
        expect(screen.getByTestId('dataIdValue').textContent).toBe('dataValue')
    })  
    test('should match snapshot', async () => {
        const tree = renderer.create(<UserDataInfo dataIdTitle='dataIdTitle' dataTitle='dataTitle' dataValue='dataValue'  dataIdValue='dataIdValue'   />).toJSON();
        expect(tree).toMatchSnapshot();
      });

})