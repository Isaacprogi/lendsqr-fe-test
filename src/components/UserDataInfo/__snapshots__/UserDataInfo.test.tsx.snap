// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test for NavBar component > should match snapshot 1`] = `
<div
  className="details"
>
  <span
    className="data-value"
    data-testid="dataIdTitle"
  >
    dataTitle
  </span>
  <span
    className="data-title"
    data-testid="dataIdValue"
  >
    dataValue
  </span>
</div>
`;

exports[`test for NavBar component should match snapshot 1`] = `
<div
  className="details"
>
  <span
    className="data-value"
    data-testid="dataIdTitle"
  >
    dataTitle
  </span>
  <span
    className="data-title"
    data-testid="dataIdValue"
  >
    dataValue
  </span>
</div>
`;
